metadataserver/auth/auth.go

```diff
@@ -0,0 +1,132 @@
+package auth
+
+import (
+	"crypto"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Netflix/titus-executor/metadataserver/identity"
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+// Authenticator generates and authenticates tokens
+type Authenticator interface {
+	GenerateToken(ttl time.Duration) (string, error)
+	VerifyToken(token string) bool
+}
+
+type envelope struct {
+	// Token token
+	Token string
+
+	// Expiration in nanoseconds
+	Expiration int64
+}
+
+// JWTAuthenticator authenticates using JWT
+type JWTAuthenticator struct {
+	Key []byte
+}
+
+// GenerateToken token
+func (a *JWTAuthenticator) GenerateToken(ttl time.Duration) (string, error) {
+	claims := &jwt.StandardClaims{
+		ExpiresAt: time.Now().Add(ttl).Unix(),
```

Comment:
Please add the task ID to the token, or similar. I think they call this "Audience" in JWT terms. I believe partially it'll improve security, but more so, improve debugging.